tests/integration/test_kube_ovn.py

```diff
@@ -306,6 +308,60 @@ async def test_multi_nic_ipam(multi_nic_ipam):
     assert ip_address(iface_addrs["net1"][0]["local"]) in ip_network("10.123.123.0/24")
 
 
+class BGPError(Exception):
+    pass
+
+
+@retry(
+    retry=retry_if_exception_type(BGPError),
+    stop=stop_after_delay(60 * 10),
+    wait=wait_fixed(1),
+    before=before_log(log, logging.INFO),
+)
+async def run_bird_curl_test(ops_test, unit, ip_to_curl):
+    retcode, stdout, stderr = await ops_test.run(
+        "juju",
+        "ssh",
+        "-m",
+        ops_test.model_full_name,
+        unit.name,
+        "curl",
+        "--connect-timeout",
+        "5",
+        ip_to_curl,
+    )
+    if retcode == 0:
+        return True
+    else:
+        raise BGPError(f"failed to reach {ip_to_curl} from {unit.name}")
+
+
+@pytest.mark.usefixtures("bird")
+async def test_bgp_pods(ops_test, annotated_nginx_pods):
+    log.info("Verifying pods are reachable from bird units ...")
+    bird_app = ops_test.model.applications["bird"]
+    for unit in bird_app.units:
+        for pod in annotated_nginx_pods:
+            assert await run_bird_curl_test(ops_test, unit, pod.status.podIP)
+
+
+@pytest.mark.usefixtures("bird")
+async def test_bgp_subnet(ops_test, annotated_subnet_nginx_pods):
+    log.info("Verifying pods are reachable from bird units via exposed subnet ...")
+    bird_app = ops_test.model.applications["bird"]
+    for unit in bird_app.units:
+        for pod in annotated_subnet_nginx_pods:
+            assert await run_bird_curl_test(ops_test, unit, pod.status.podIP)
+
+
+@pytest.mark.usefixtures("bird")
+async def test_bgp_service(ops_test, nginx_cluster_ip):
+    log.info("Verifying service is reachable from bird units ...")
+    bird_app = ops_test.model.applications["bird"]
+    for unit in bird_app.units:
+        assert await run_bird_curl_test(ops_test, unit, nginx_cluster_ip)
```

Comment:
these three tests look VERY similar.  I'm not saying you should -- but would [pytest parameterized fixtures](https://docs.pytest.org/en/6.2.x/fixture.html#fixture-parametrize) help make this one test rather than three?

Reply:
I will add the parameterized fixtures and knock this down to 1 tests

Reply:
added a parameterized fixture that annotates the various resources and reduces this test down to 1 test function